synapse/federation/federation_client.py

```diff
@@ -1113,14 +1114,17 @@ async def get_public_rooms(
                 requests over federation
 
         """
-        return await self.transport_layer.get_public_rooms(
-            remote_server,
-            limit,
-            since_token,
-            search_filter,
-            include_all_networks=include_all_networks,
-            third_party_instance_id=third_party_instance_id,
-        )
+        try:
+            return await self.transport_layer.get_public_rooms(
+                remote_server,
+                limit,
+                since_token,
+                search_filter,
+                include_all_networks=include_all_networks,
+                third_party_instance_id=third_party_instance_id,
+            )
+        except (RequestSendFailed, HttpResponseException):
+            raise SynapseError(502, "Failed to fetch room list")
```

Comment:
Don't we want to log the error as well?

Reply:
We'll have already logged that the federation request has failed, so I don't think there's much point